### PR TITLE
MYST-376 Install Jpex DI container

### PR DESCRIPTION
**Pull request by shroomist**
proposals now fetched only when we click dropdown

### CODE REVIEW

**Comment by shroomist** [comment]
@Waldz you're not pushing your changes, I cant test this

File: src/renderer/dependencies/modules/vue.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// @flow
+import type {Container} from '../../../app/di'
+import Vue from 'vue'
+import axios from 'axios'
+import App from '../../App'
+import storeFactory from '../../store/factory'
+import routerFactory from '../../router/factory'
+
+function bootstrap (container: Container) {
+  container.service(
```
Comment (shroomist) [inline src/renderer/dependencies/modules/vue.js]: Why do we need to contain our frontend application entry point in DI, I couldn't find where we're using it.

File: src/renderer/store/factory.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import terms from './modules/terms'
+import errors from './modules/errors'
+import identityFactory from './modules/identity'
+
+function factory (tequilapi, rendererCommunication) {
+  Vue.use(Vuex)
+
+  return new Vuex.Store({
+    modules: {
+      main: mainFactory(tequilapi),
```
Comment (shroomist) [inline src/renderer/store/factory.js]: do have an intention to use DI to pass tequilapi to the modules, or do we stick to this Factory approach?

File: src/renderer/dependencies/modules/application.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// @flow
```
Comment (ignasbernotas) [inline src/renderer/dependencies/modules/application.js]: why is this file called application if it registers the communication service?:)

File: src/renderer/dependencies/modules/vue.js (hunk 1 above)
Comment (Waldz) [reply src/renderer/dependencies/modules/vue.js]: I'm seeking `main.js` to become minimal (to be just a bootstrap file)
e.g. `new MainWindowApplication().start()` in future, no hooks, no more gluing in `main.js`

File: src/main/dependencies/modules/application.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// @flow
+import type {Container} from '../../../app/di'
+
+function bootstrap (container: Container) {}
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/main/dependencies/modules/application.js]: Just a skeleton/placeholder, for team to register another dependencies

File: src/renderer/components/CountrySelect.vue [hunk 5]
```diff
@@ -87,10 +80,26 @@
...
+          error.original = e
+          this.$store.commit(type.SHOW_ERROR, error)
+          bugReporter.renderer.captureException(error)
+        }
+
+        this.countriesIsLoading = false
       }
+    },
+    mounted () {
+      tequilapi = tequilAPI()
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/renderer/components/CountrySelect.vue]: Fixed

File: src/renderer/components/CountrySelect.vue [hunk 6]
```diff
@@ -87,10 +78,20 @@
...
+      async fetchCountries () {
+        this.countriesIsLoading = true
+
+        let response = await tequilapi.proposal.list()
+        this.countriesList = response.proposals.map(proposalToCountry)
+
+        this.countriesIsLoading = false
       }
+    },
+    mounted () {
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/renderer/components/CountrySelect.vue]: Fixed

File: src/app/di/vue-container.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+// @flow
+import type {Container, ServiceFactory} from './index'
+import Vue from 'vue'
+import Injector from 'vue-inject'
+
+class VueContainer implements Container {
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/app/di/vue-container.js]: Described DI classes and interfaces

File: src/renderer/components/CountrySelect.vue [hunk 8]
```diff
@@ -87,10 +78,26 @@
         }
 
         return path.join('static', 'flags', code + '.svg')
+      },
+      async fetchCountries () {
+        this.countriesIsLoading = true
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/renderer/components/CountrySelect.vue]: Fixed in #128

File: src/renderer/components/CountrySelect.vue [hunk 9]
```diff
@@ -87,10 +78,26 @@
...
+        let response
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/renderer/components/CountrySelect.vue]: Was fixed already in #128

File: src/renderer/components/CountrySelect.vue [hunk 10]
```diff
@@ -87,10 +78,26 @@
...
+
+        try {
+          response = await this.tequilapi.proposal.list()
+          this.countriesList = response.proposals.map(proposalToCountry)
+        } catch (e) {
+          console.log('Proposal fetching in initialization failed', e)
```
[the remark this answers is not in the document]
Comment (Waldz) [reply src/renderer/components/CountrySelect.vue]: Was fixed already in #128

File: test/unit/specs/pages/VpnLoader.spec.js [hunk 11]
```diff
@@ -21,45 +21,50 @@ import messages from '../../../../src/app/messages'
...
-
-  return vm
-}
-
 describe('VpnLoader', () => {
+  let tequilapi
   let mock
   let clock
 
+  async function mountComponent (tequilapi) {
```
Comment (ignasbernotas) [inline test/unit/specs/pages/VpnLoader.spec.js]: Please add todo o to use `createLocalVue` from '@vue/test-utils'

Comment (Waldz) [reply test/unit/specs/pages/VpnLoader.spec.js]: Added

**Pull request merged**